io_scene_nif/modules/nif_export/property/texture/__init__.py

```diff
@@ -71,13 +71,37 @@ def _reset_fields(self):
                 "Specular": None,
                 "Normal": None,
         }
+    
+    def get_input_node_of_type(self, input_socket, node_types):
+        #search back in the node tree for nodes of a certain type(s), depth-first
+        links = input_socket.links
+        if not links:
+            #this socket has no inputs
+            return None
+        node = links[0].from_node
+        if isinstance(node, node_types):
+            #the input node is of the required type
+            return node
+        else:
+            if len(node.inputs) > 0:
+                for input in node.inputs:
+                    #check every input if somewhere up that tree is a node of the required type
+                    input_results = self.get_input_node_of_type(input, node_types)
+                    if input_results:
+                        return input_results
+                #we found nothing
+                return None
+            else:
+                #this has no inputs, and doesn't classify itself
+                return None
 
     def get_uv_node(self, b_texture_node):
-        # check if a node is plugged into the b_texture_node's vector input
-        links = b_texture_node.inputs[0].links
-        if not links:
-            return 0
-        uv_node = links[0].from_node
+        uv_node = self.get_input_node_of_type(b_texture_node.inputs[0], (bpy.types.ShaderNodeUVMap, bpy.types.ShaderNodeTexCoord))
+        if uv_node is None:
+            links = b_texture_node.inputs[0].links
+            if not links:
+                #nothing is plugged in, so it will use the first UV map
```

Comment:
Should we warn the user

Reply:
That's how it was before. It also reflects how blender renders them (when the socket has no inputs, it picks the first UV map), so I don't think it is unexpected behaviour, it just removes the requirement to link to the UV node.

Reply:
Its something we should look out for if users get confuzzled why things are working.